--- binutils-2.9.1/configure	Fri May  1 08:48:46 1998
+++ binutils-2.9.1/configure	Sun Aug 23 00:00:00 1998
@@ -34,8 +34,8 @@
 export PATH || (echo "OOPS, this isn't sh.  Desperation time.  I will feed myself to sh."; sh $0 $argv; kill $$)
 
 remove=rm
-hard_link=ln
-symbolic_link='ln -s'
+hard_link=cp
+symbolic_link=cp
 
 #for Test
 #remove="echo rm"
@@ -64,7 +64,10 @@ norecursion=
 other_options=
 package_makefile_frag=
 package_makefile_rules_frag=
-prefix=/usr/local
+# This is a hack so we don't have to remember to always configure using
+# the "--prefix=/gg" option.  Personally I feel this should be settable
+# in a config file.  -fnf
+prefix=/gg
 progname=
 program_prefix=
 program_prefixoption=
@@ -144,9 +147,12 @@ NO_EDIT="This file was generated automat
 ## path.  Since PATH might include "." we also add `pwd` to the end of PATH.
 ##
 
-progname=$0
+# For some reason, the value of $0 in AmigaOS pdksh ends up with a
+# trailing '/' that needs to be stripped.
+progname=`echo $0 | sed 's:/$::'`
 # if PWD already has a value, it is probably wrong.
-if [ -n "$PWD" ]; then PWD=`pwd`; fi
+# The AmigaOS pksh can't change the value of PWD, it's readonly.
+#if [ -n "$PWD" ]; then PWD=`pwd`; fi
 
 case "${progname}" in
 /*) ;;
@@ -964,18 +970,76 @@ else
       fi
     done
     IFS="$save_ifs"
-    CC=${CC-cc}
+  fi
+  # If CC is still not set, try to get bcc.
+  if [ -z "${CC}" ]; then
+    IFS="${IFS= 	}"; save_ifs="$IFS"; IFS="${IFS}:"
+    for dir in $PATH; do
+      test -z "$dir" && dir=.
+      if test -f $dir/bcc; then
+	CC="bcc"
+	echo 'void f(){}' > conftest.c
+	if test -z "`${CC} -g -c conftest.c 2>&1`"; then
+	  CFLAGS=${CFLAGS-"-g -O2"}
+	  CXXFLAGS=${CFLAGS-"-g -O2"}
+	else
+	  CFLAGS=${CFLAGS-"-O2"}
+	  CXXFLAGS=${CFLAGS-"-O2"}
+	fi
+	rm -f conftest*
+	break
+      fi
+    done
+    IFS="$save_ifs"
+  fi
+  # If CC is still not set, try to get mwcc.
+  if [ -z "${CC}" ]; then
+    IFS="${IFS= 	}"; save_ifs="$IFS"; IFS="${IFS}:"
+    for dir in $PATH; do
+      test -z "$dir" && dir=.
+      if test -f $dir/mwcc; then
+	CC="mwcc -I- -I. -I${prefix}/include -L${prefix}/lib -opt global -nodup"
+	echo 'void f(){}' > conftest.c
+	if test -z "`${CC} -g -c conftest.c 2>&1`"; then
+	  CFLAGS=${CFLAGS-"-g -O2"}
+	  CXXFLAGS=${CFLAGS-"-g -O2"}
+	else
+	  CFLAGS=${CFLAGS-"-O2"}
+	  CXXFLAGS=${CFLAGS-"-O2"}
+	fi
+	rm -f conftest*
+	break
+      fi
+    done
+    IFS="$save_ifs"
   fi
 
+  CC=${CC-gcc}
   CXX=${CXX-"c++"}
   CFLAGS=${CFLAGS-"-g"}
   CXXFLAGS=${CXXFLAGS-"-g -O2"}
 fi
 
+case "$CC" in
+	*mwcc* | *bcc* )
+		AR="mwld"
+		ARFLAGS="-xml -o"
+		RANLIB="true"
+		;;
+	*)
+		AR=${AR-"ar"}
+		ARFLAGS=${ARFLAGS-"cr"}
+		RANLIB=${RANLIB-"ranlib"}
+		;;
+esac
+
 export CC
 export CXX
 export CFLAGS
 export CXXFLAGS
+export AR
+export ARFLAGS
+export RANLIB
 
 # FIXME: This should be in configure.in, not configure
 case "$host" in
@@ -983,6 +1047,8 @@ case "$host" in
 	    enable_gdbtk=no ;;
 	*msdosdjgpp*)
 	    enable_gdbtk=no ;;
+	*beos*)
+	    enable_gdbtk=no ;;
 esac
 
 # FIXME: This should be in configure.in, not configure
@@ -1305,6 +1371,9 @@ EOF
 			s%^CXXFLAGS[ 	]*=.*$%CXXFLAGS = ${CXXFLAGS}%
 			}" \
 		    -e "s|^SHELL[	 ]*=.*$|SHELL = ${config_shell}|" \
+		    -e "s:^AR[	 ]*=.*$:AR = ${AR}:" \
+		    -e "s:^ARFLAGS[	 ]*=.*$:ARFLAGS = ${ARFLAGS}:" \
+		    -e "s:^RANLIB[	 ]*=.*$:RANLIB = ${RANLIB}:" \
                     -e "s|^srcdir[ 	]*=.*$|srcdir = ${makesrcdir}|" \
                     -e "s///" \
                     -e "s:^program_prefix[ 	]*=.*$:program_prefix = ${program_prefix}:" \
@@ -1468,7 +1537,9 @@ if [ -z "${norecursion}" -a -n "${config
 
 ### The recursion line is here.
 			if [ ! -z "${recprog}" ] ; then
-	                        if eval ${config_shell} ${recprog} ${verbose} ${buildopt} --host=${host_alias} --target=${target_alias} \
+	                        	echo ${config_shell} ${recprog} ${verbose} ${buildopt} --host=${host_alias} --target=${target_alias} ${prefixoption} ${tmpdiroption} ${exec_prefixoption} ${srcdiroption} ${program_prefixoption} ${program_suffixoption} \
+					${program_transform_nameoption} ${site_option} ${withoptions} ${withoutoptions} ${enableoptions} ${disableoptions} ${removing} ${other_options} ${redirect}
+					if eval ${config_shell} ${recprog} ${verbose} ${buildopt} --host=${host_alias} --target=${target_alias} \
         	                        ${prefixoption} ${tmpdiroption} ${exec_prefixoption} \
                 	                ${srcdiroption} ${diroptions} ${program_prefixoption} ${program_suffixoption} ${program_transform_nameoption} ${site_option} ${withoptions} ${withoutoptions} ${enableoptions} ${disableoptions} ${floating_pointoption} ${cache_file_option} ${removing} ${other_options} ${redirect} ; then
 	                                true
