--- gcc-3.4.6/gcc/c-incpath.c	2013-05-19 19:56:46.000000000 +0200
+++ gcc-3.4.6-patched/gcc/c-incpath.c	2013-05-19 20:23:32.000000000 +0200
@@ -28,14 +28,14 @@
 #include "c-incpath.h"
 #include "cppdefault.h"
 
-/* Windows does not natively support inodes, and neither does MSDOS.
+/* Windows does not natively support inodes, and neither does MSDOS or AmigaOS.
    Cygwin's emulation can generate non-unique inodes, so don't use it.
    VMS has non-numeric inodes.  */
 #ifdef VMS
 # define INO_T_EQ(A, B) (!memcmp (&(A), &(B), sizeof (A)))
 # define INO_T_COPY(DEST, SRC) memcpy(&(DEST), &(SRC), sizeof (SRC))
 #else
-# if (defined _WIN32 && ! defined (_UWIN)) || defined __MSDOS__
+# if (defined _WIN32 && ! defined (_UWIN)) || defined __MSDOS__ || __amigaos__
 #  define INO_T_EQ(A, B) 0
 # else
 #  define INO_T_EQ(A, B) ((A) == (B))
