--- gcc-3.4.6/gcc/c-tree.h	2004-03-01 00:34:49.000000000 +0100
+++ gcc-3.4.6-patched/gcc/c-tree.h	2013-05-19 20:23:32.000000000 +0200
@@ -214,7 +214,9 @@
 extern void pending_xref_error (void);
 extern void c_push_function_context (struct function *);
 extern void c_pop_function_context (struct function *);
-extern void push_parm_decl (tree);
+/* begin-GG-local: explicit register specification for parameters */
+extern void push_parm_decl (tree, tree);
+/* end-GG-local */
 extern tree pushdecl_top_level (tree);
 extern void pushtag (tree, tree);
 extern tree set_array_declarator_type (tree, tree, int);
