--- fd2sfd-1.0/fd2inline.c.orig	2003-07-28 10:42:50.000000000 +0200
+++ fd2sfd-1.0/fd2inline.c	2014-01-06 10:29:56.000000000 +0100
@@ -406,7 +408,7 @@
 
    if (fname)
    {
-      result=malloc(sizeof(fdFile));
+      result=calloc(1,sizeof(fdFile));
       if (result)
       {
 	 result->file=fopen(fname, "r");
@@ -722,7 +724,7 @@
    fdDef *result;
    regs count;
 
-   result=malloc(sizeof(fdDef));
+   result=calloc(1,sizeof(fdDef));
 
    if (result)
    {
@@ -2069,13 +2071,13 @@
       return EXIT_FAILURE;
    }
 
-   if (!(arrdefs=malloc(FDS*sizeof(fdDef*))))
+   if (!(arrdefs=calloc(FDS,sizeof(fdDef*))))
    {
       fprintf(stderr, "No mem for FDs\n");
       rc = EXIT_FAILURE;
       goto quit;
    }
-   if (!(arrcmts=malloc(FDS*sizeof(char*))))
+   if (!(arrcmts=calloc(FDS,sizeof(char*))))
    {
       fprintf(stderr, "No mem for FD comments\n");
       rc = EXIT_FAILURE;
@@ -2126,7 +2128,7 @@
    /* Make a copy before we sort, since we need to process the
       definitions in bias order, not lexical order */
 
-   if (!(defs=malloc(fds*sizeof(fdDef*))))
+   if (!(defs=calloc(fds,sizeof(fdDef*))))
    {
       fprintf(stderr, "No mem for FDs\n");
       rc = EXIT_FAILURE;
