--- fd2sfd-1.0/Makefile.in	2003-07-31 08:10:19.000000000 +0200
+++ fd2sfd-1.0/Makefile.in	2014-05-29 00:09:00.000000000 +0200
@@ -17,6 +17,7 @@
 
 INSTALL		= @INSTALL@
 INSTALL_PROGRAM	= @INSTALL_PROGRAM@ -s
+INSTALL_SCRIPT	= @INSTALL_SCRIPT@
 INSTALL_DATA	= @INSTALL_DATA@
 
 VPATH		= $(srcdir)
@@ -40,49 +41,11 @@
 
 	# Install tools
 	$(INSTALL_PROGRAM) $(EXECUTABLE) $(bindir)
-	$(INSTALL_PROGRAM) cross/bin/gg-fix-includes $(bindir)
+	$(INSTALL_SCRIPT) cross/bin/gg-fix-includes $(bindir)
 	for f in `cd $(srcdir)/cross/share/ && find * -type f`; do \
-	  $(INSTALL_DATA) -D $(srcdir)/cross/share/$${f} $(fd2sfddir)/$${f}; \
-	done
-
-	# Install example file structure for FDs
-	$(INSTALL) -d $(prefix)/os-lib/fd/3rd
-	$(INSTALL) -d $(prefix)/os-lib/fd/3rd-amigaos
-	$(INSTALL) -d $(prefix)/os-lib/fd/3rd-amithlon
-	$(INSTALL) -d $(prefix)/os-lib/fd/3rd-morphos
-	$(INSTALL) -d $(prefix)/os-lib/fd/amigaos
-	$(INSTALL) -d $(prefix)/os-lib/fd/morphos
-	$(INSTALL_DATA) $(srcdir)/README.os-lib.fd $(prefix)/os-lib/fd/README
-
-	# Install example file structure for SFDs
-	$(INSTALL) -d $(prefix)/os-lib/sfd/3rd
-	$(INSTALL) -d $(prefix)/os-lib/sfd/3rd-amigaos
-	$(INSTALL) -d $(prefix)/os-lib/sfd/3rd-amithlon
-	$(INSTALL) -d $(prefix)/os-lib/sfd/3rd-morphos
-	$(INSTALL) -d $(prefix)/os-lib/sfd/amigaos
-	$(INSTALL) -d $(prefix)/os-lib/sfd/morphos
-	$(INSTALL_DATA) $(srcdir)/README.os-lib.sfd $(prefix)/os-lib/sfd/README
-
-	# Install example file structure for includes
-	$(INSTALL) -d $(prefix)/os-include/3rd
-	$(INSTALL) -d $(prefix)/os-include/3rd-amigaos
-	$(INSTALL) -d $(prefix)/os-include/3rd-amithlon
-	$(INSTALL) -d $(prefix)/os-include/3rd-morphos
-	for dir in classes clib datatypes devices diskfont dos exec \
-		   gadgets graphics hardware images intuition libraries \
-		   prefs reaction resources rexx \
-		   utility workbench; do \
-	  $(INSTALL) -d $(prefix)/os-include/amigaos/$${dir}; \
-	done
-
-	for dir in classes clib cybergraphx datatypes devices diskfont dos \
-		   emul exec gadgets graphics guigfx hardware images \
-		   intuition libraries mui net ppcdisslib ppclib \
-		   prefs reaction render resources rexx \
-		   utility workbench; do \
-	  $(INSTALL) -d $(prefix)/os-include/morphos/$${dir}; \
+	  mkdir -p `dirname $(fd2sfddir)/$${f}`; \
+	  $(INSTALL_DATA) $(srcdir)/cross/share/$${f} `dirname $(fd2sfddir)/$${f}`; \
 	done
-	$(INSTALL_DATA) $(srcdir)/README.os-include $(prefix)/os-include/README
 
 tgz:		all gg-fd2sfd.spec
 	cd $(srcdir) && cvs diff >/dev/null || (echo "Not checked in!"; exit 10)
