--- binutils-2.14/gas/configure.in	2003-06-02 22:35:23.000000000 +0200
+++ binutils-2.14-patched/gas/configure.in	2013-04-24 18:37:53.000000000 +0200
@@ -326,6 +328,8 @@
       m68hc11-*-* | m6811-*-*)		fmt=elf ;;
       m68hc12-*-* | m6812-*-*)		fmt=elf ;;
 
+      m68k-*-amigaoshunk)		fmt=amigahunk em=amiga bfd_gas=yes ;;
+      m68k-*-amigaos*)			fmt=aout em=amiga ;;
       m68k-*-vxworks*)			fmt=aout em=sun3 ;;
       m68k-ericsson-ose)		fmt=aout em=sun3 ;;
       m68k-*-sunos*)			fmt=aout em=sun3 ;;
@@ -895,7 +901,7 @@
 
 case ${primary_bfd_gas}-${target_cpu_type}-${obj_format} in
   yes-*-coff)	need_bfd=yes ;;
-  no-*-coff)	need_bfd=yes
+  no-*-coff | yes-m68k-amigahunk | no-m68k-amigahunk)	need_bfd=yes
 		AC_DEFINE(MANY_SEGMENTS, 1, [old COFF support?]) ;;
 esac
 
