--- gcc-2.95.3/gcc/config/rs6000/rs6000.h	Wed Mar 22 19:54:05 2000
+++ gcc/gcc/config/rs6000/rs6000.h	Thu Sep 27 19:35:54 2001
@@ -3276,6 +3276,7 @@
 extern int current_file_function_operand ();
 extern int input_operand ();
 extern int small_data_operand ();
+extern int baserel_data_operand ();
 extern void init_cumulative_args ();
 extern void function_arg_advance ();
 extern int function_arg_boundary ();
