--- binutils-2.14/ld/configure.tgt	2003-06-12 16:25:52.000000000 +0200
+++ binutils-2.14-patched/ld/configure.tgt	2013-04-24 18:37:52.000000000 +0200
@@ -340,6 +341,7 @@
 m68k-sony-*)		targ_emul=news ;;
 m68k-hp-bsd*)		targ_emul=hp300bsd ;;
 m68*-motorola-sysv*)	targ_emul=delta68 ;;
+m68*-*-amigaos*)	targ_emul=amiga ; targ_extra_emuls=amiga_bss;;
 m68*-*-aout)		targ_emul=m68kaout ;;
 m68*-*-coff)		targ_emul=m68kcoff ;;
 m68*-*-elf)		targ_emul=m68kelf ;;
