--- gcc-2.95.3/gcc/cp/config-lang.in	1999-01-06 21:57:30.000000000 +0100
+++ gcc-2.95.3/gcc/cp/config-lang.in	2012-08-04 11:53:28.000000000 +0200
@@ -39,3 +39,8 @@
 lib2funcs=cplib2.txt
 
 outputs=cp/Makefile
+
+# GG local hack.  Need this so that exception handling stuff gets
+# built, otherwise cannot bootstrap ADA.  Since we compile with gcc
+# on every platform we care about, this is no problem.
+boot_language=yes
