--- m68k-amigaos-ahi/Developer/Include/C/devices/ahi.h~	2005-11-20 23:39:44.000000000 +0200
+++ m68k-amigaos-ahi/Developer/Include/C/devices/ahi.h	2017-07-20 12:51:00.000000000 +0300
@@ -118,7 +118,11 @@
 {
 	ULONG	ahie_Effect;			/* Set to AHIET_DSPMASK */
 	UWORD	ahiedm_Channels;		/* Number of elements in array */
+#if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L)
+	UBYTE	ahiedm_Mask[];			/* Here follows the array */
+#else
 	UBYTE	ahiedm_Mask[0];			/* Here follows the array */
+#endif
 };
 
 #define AHIEDM_WET		(0)
@@ -145,7 +149,11 @@
 	UWORD		 ahieci_Channels;
 	UWORD		 ahieci_Pad;
  /* The rest is filled by AHI */
+#if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L)
+ 	ULONG		 ahieci_Offset[];	/* The array follows */
+#else
  	ULONG		 ahieci_Offset[0];	/* The array follows */
+#endif
 };
 
 /*** TAGS */
