--- binutils-2.9.1/gas/read.h	1998-05-01 17:45:16.000000000 +0200
+++ binutils-2.9.1/gas/read.h	2012-08-04 12:43:35.000000000 +0200
@@ -117,6 +117,9 @@
 extern void s_app_line PARAMS ((int));
 extern void s_comm PARAMS ((int));
 extern void s_data PARAMS ((int));
+#ifdef OBJ_AMIGAHUNK
+extern void s_data_chip PARAMS ((int));
+#endif
 extern void s_desc PARAMS ((int));
 extern void s_else PARAMS ((int arg));
 extern void s_end PARAMS ((int arg));
