--- gcc-2.95.3/gcc/ginclude/varargs.h	1999-07-07 02:19:05.000000000 +0200
+++ gcc-2.95.3/gcc/ginclude/varargs.h	2012-08-04 11:53:28.000000000 +0200
@@ -194,7 +194,7 @@
 /* Michael Eriksson <mer@sics.se> at Thu Sep 30 11:00:57 1993:
    Sequent defines _VA_LIST_ in <machine/machtypes.h> to be the type to
    use for va_list (``typedef _VA_LIST_ va_list'') */
-#if !defined (_VA_LIST_) || defined (__BSD_NET2__) || defined (____386BSD____) || defined (__bsdi__) || defined (__sequent__) || defined (__FreeBSD__) || defined(WINNT)
+#if !defined (_VA_LIST_) || defined (__BSD_NET2__) || defined (____386BSD____) || defined (__bsdi__) || defined (__sequent__) || defined (__FreeBSD__) || defined(WINNT) || defined(__amigaos__)
 /* The macro _VA_LIST_DEFINED is used in Windows NT 3.5  */
 #ifndef _VA_LIST_DEFINED
 /* The macro _VA_LIST is used in SCO Unix 3.2.  */
@@ -205,7 +205,7 @@
 #endif /* not _VA_LIST_T_H */
 #endif /* not _VA_LIST */
 #endif /* not _VA_LIST_DEFINED */
-#if !(defined (__BSD_NET2__) || defined (____386BSD____) || defined (__bsdi__) || defined (__sequent__) || defined (__FreeBSD__))
+#if !(defined (__BSD_NET2__) || defined (____386BSD____) || defined (__bsdi__) || defined (__sequent__) || defined (__FreeBSD__) || defined(__amigaos__))
 #define _VA_LIST_
 #endif
 #ifndef _VA_LIST
