--- binutils-2.10.1/bfd/Makefile.in	2000-04-12 01:58:39.000000000 +0200
+++ binutils-2.10.1/bfd/Makefile.in	2012-08-08 12:35:51.000000000 +0200
@@ -230,7 +230,10 @@
 # The .o files needed by all of the 32 bit vectors that are configured into
 # target_vector in targets.c if configured with --enable-targets=all.
 BFD32_BACKENDS = \
+	amigaos.lo \
+	amigaoslink.lo \
 	aout-adobe.lo \
+	aout-amiga.lo \
 	aout-arm.lo \
 	aout-ns32k.lo \
 	aout-sparcle.lo \
@@ -361,7 +364,10 @@
 
 
 BFD32_BACKENDS_CFILES = \
+	amigaos.c \
+	amigaoslink.c \
 	aout-adobe.c \
+	aout-amiga.c \
 	aout-arm.c \
 	aout-ns32k.c \
 	aout-sparcle.c \
@@ -563,7 +569,7 @@
 SOURCE_HFILES = \
 	aout-target.h aoutf1.h aoutx.h coffcode.h coffswap.h \
 	ecoffswap.h elf32-arm.h elfcode.h elfcore.h \
-	elflink.h freebsd.h genlink.h go32stub.h libaout.h \
+	elflink.h freebsd.h genlink.h go32stub.h libamiga.h libaout.h \
 	libbfd.h libcoff.h libecoff.h elf-bfd.h libhppa.h libieee.h \
 	libnlm.h liboasys.h libpei.h netbsd.h nlm-target.h nlmcode.h \
 	nlmswap.h ns32k.h peicode.h som.h vms.h
@@ -1248,6 +1254,9 @@
 cpu-we32k.lo: cpu-we32k.c
 cpu-w65.lo: cpu-w65.c
 cpu-z8k.lo: cpu-z8k.c
+amigaos.lo: amigaos.c libamiga.h bfd.h
+amigaoslink.lo: bfd.h libamiga.h amigaoslink.c
+aout-amiga.lo: aout-amiga.c aoutf1.h bfd.h
 aout-adobe.lo: aout-adobe.c $(INCDIR)/aout/adobe.h \
   $(INCDIR)/aout/stab_gnu.h $(INCDIR)/aout/stab.def libaout.h \
   $(INCDIR)/bfdlink.h
