--- binutils-2.10.1/gas/tc.h	1999-05-03 09:28:41.000000000 +0200
+++ binutils-2.10.1/gas/tc.h	2012-08-08 12:35:51.000000000 +0200
@@ -23,25 +23,6 @@
 
 extern const pseudo_typeS md_pseudo_table[];
 
-/* JF moved this here from as.h under the theory that nobody except MACHINE.c
-   and write.c care about it anyway. */
-
-struct relax_type
-{
-  /* Forward reach. Signed number. > 0. */
-  long rlx_forward;
-  /* Backward reach. Signed number. < 0. */
-  long rlx_backward;
-
-  /* Bytes length of this address. */
-  unsigned char rlx_length;
-
-  /* Next longer relax-state.  0 means there is no 'next' relax-state. */
-  relax_substateT rlx_more;
-};
-
-typedef struct relax_type relax_typeS;
-
 extern const int md_reloc_size;	/* Size of a relocation record */
 
 char *md_atof PARAMS ((int what_statement_type, char *literalP, int *sizeP));
