--- binutils-2.10.1/gas/write.h	1999-05-03 09:28:41.000000000 +0200
+++ binutils-2.10.1/gas/write.h	2012-08-08 12:35:51.000000000 +0200
@@ -189,17 +189,18 @@
 #ifdef BFD_ASSEMBLER
 extern fixS *fix_new
   PARAMS ((fragS * frag, int where, int size, symbolS * add_symbol,
-	   offsetT offset, int pcrel, bfd_reloc_code_real_type r_type));
+	   offsetT offset, int pcrel, bfd_reloc_code_real_type r_type,
+	   int baserel));
 extern fixS *fix_new_exp
   PARAMS ((fragS * frag, int where, int size, expressionS *exp, int pcrel,
-	   bfd_reloc_code_real_type r_type));
+	   bfd_reloc_code_real_type r_type, int baserel));
 #else
 extern fixS *fix_new
   PARAMS ((fragS * frag, int where, int size, symbolS * add_symbol,
-	   offsetT offset, int pcrel, int r_type));
+	   offsetT offset, int pcrel, int r_type, int baserel));
 extern fixS *fix_new_exp
   PARAMS ((fragS * frag, int where, int size, expressionS *exp, int pcrel,
-	   int r_type));
+	   int r_type, int baserel));
 #endif
 
 extern void write_print_statistics PARAMS ((FILE *));
