--- gcc-2.95.3/config.sub	1999-08-04 10:09:26.000000000 +0200
+++ gcc-2.95.3/config.sub	2012-08-04 11:53:28.000000000 +0200
@@ -277,14 +277,14 @@
 		os=-sysv
 		;;
 	amiga | amiga-*)
-		basic_machine=m68k-cbm
+		basic_machine=m68k-unknown
 		;;
 	amigaos | amigados)
-		basic_machine=m68k-cbm
+		basic_machine=m68k-unknown
 		os=-amigaos
 		;;
 	amigaunix | amix)
-		basic_machine=m68k-cbm
+		basic_machine=m68k-unknown
 		os=-sysv4
 		;;
 	apollo68)
