--- binutils-2.9.1/bfd/doc/bfd.texinfo	1998-05-01 17:48:02.000000000 +0200
+++ binutils-2.9.1/bfd/doc/bfd.texinfo	2012-08-04 12:43:35.000000000 +0200
@@ -302,9 +302,10 @@
 @chapter BFD back ends
 @menu
 * What to Put Where::
-* aout ::	a.out backends
-* coff ::	coff backends
-* elf  ::	elf backends
+* aout  ::	a.out backends
+* coff  ::	coff backends
+* elf   ::	elf backends
+* amiga ::      amigaos backend
 @ignore
 * oasys ::	oasys backends
 * ieee ::	ieee backend
@@ -320,11 +321,15 @@
 @node coff, elf, aout, BFD back ends
 @include  coffcode.texi
 
-@node elf,  , coff, BFD back ends
+@node elf, amiga  , coff, BFD back ends
 @include  elf.texi
 @c Leave this out until the file has some actual contents...
 @c @include  elfcode.texi
 
+@node amiga,  , elf, BFD back ends
+@include amiga.texi
+@include amigalink.texi
+
 @node Index,  , BFD back ends , Top
 @unnumbered Index
 @printindex cp
