--- binutils-2.9.1/include/coff/internal.h	Fri May  1 08:48:27 1998
+++ binutils-2.9.1/include/coff/internal.h	Sun Aug 23 00:00:00 1998
@@ -92,6 +92,10 @@ typedef struct _IMAGE_DATA_DIRECTORY 
 #define NT_EXE_IMAGE_BASE 0x400000
 #define NT_DLL_IMAGE_BASE 0x10000000
 
+/* default image base for BeOS */
+#define BEOS_EXE_IMAGE_BASE 0x80000000
+#define BEOS_DLL_IMAGE_BASE 0x80000000
+
 /* Extra stuff in a PE aouthdr */
 
 #define PE_DEF_SECTION_ALIGNMENT 0x1000
