--- binutils-2.10.1/binutils/objcopy.c	2000-09-05 10:56:22.000000000 +0200
+++ binutils-2.10.1/binutils/objcopy.c	2012-08-08 12:35:51.000000000 +0200
@@ -1407,7 +1407,8 @@
   if (relsize < 0)
     RETURN_NONFATAL (bfd_get_filename (ibfd));
 
-  if (relsize == 0)
+  /* Never, ever, strip reloc data on the Amiga! */
+  if (relsize == 0 && bfd_get_flavour(obfd) != bfd_target_amiga_flavour)
     bfd_set_reloc (obfd, osection, (arelent **) NULL, 0);
   else
     {
