--- binutils-2.9.1/configure.in	Fri May  1 09:00:42 1998
+++ binutils-2.9.1/configure.in	Sun Aug 23 00:00:00 1998
@@ -179,6 +179,9 @@ case "${host}" in
   i[3456]86-*-msdosdjgpp*)
     host_makefile_frag="${host_makefile_frag} config/mh-go32"
     ;;
+  *-*-amigaos*)
+    host_makefile_frag="${host_makefile_frag} config/mh-amigaos"
+    ;;
   *-cygwin32*)
     host_makefile_frag="${host_makefile_frag} config/mh-cygwin32"
     ;;
