--- binutils-2.9.1/Makefile.in	Fri May  1 08:52:55 1998
+++ binutils-2.9.1/Makefile.in	Sun Aug 23 00:00:00 1998
@@ -20,7 +20,7 @@
 
 srcdir = .
 
-prefix = /usr/local
+prefix = @prefix@
 exec_prefix = $(prefix)
 
 bindir=${exec_prefix}/bin
@@ -49,6 +49,9 @@ man6dir = $(mandir)/man6
 man7dir = $(mandir)/man7
 man8dir = $(mandir)/man8
 man9dir = $(mandir)/man9
+infodir = $(prefix)/info
+guidedir =$(prefix)/guide
+includedir = $(prefix)/include
 GDB_NLM_DEPS = 
 
 SHELL = /bin/sh
@@ -65,9 +68,9 @@ INSTALL_DATA = $(INSTALL) -m 644
 INSTALL_DOSREL = install-dosrel-fake
 
 AS = as
-AR = ar
-AR_FLAGS = rc
-CC = cc
+AR = @AR@
+ARFLAGS = @ARFLAGS@
+CC = @CC@
 
 # Special variables passed down in EXTRA_GCC_FLAGS.  They are defined
 # here so that they can be overridden by Makefile fragments.
@@ -93,7 +96,7 @@ LIBCXXFLAGS = $(CXXFLAGS) -fno-implicit-
 CXXFLAGS_FOR_TARGET = $(CXXFLAGS)
 LIBCXXFLAGS_FOR_TARGET = $(CXXFLAGS_FOR_TARGET) -fno-implicit-templates
 
-RANLIB = ranlib
+RANLIB = @RANLIB@
 
 DLLTOOL = dlltool
 WINDRES = windres
@@ -318,7 +321,7 @@ NM_FOR_TARGET = ` \
 # Flags to pass down to all sub-makes.
 # Please keep these in alphabetical order.
 BASE_FLAGS_TO_PASS = \
-	"AR_FLAGS=$(AR_FLAGS)" \
+	"ARFLAGS=$(ARFLAGS)" \
 	"AR_FOR_TARGET=$(AR_FOR_TARGET)" \
 	"AS_FOR_TARGET=$(AS_FOR_TARGET)" \
 	"BISON=$(BISON)" \
@@ -854,7 +857,9 @@ DO_X = \
 	do-distclean \
 	do-dvi \
 	do-info \
+	do-guide \
 	do-install-info \
+	do-install-guide \
 	do-installcheck \
 	do-mostlyclean \
 	do-maintainer-clean \
@@ -879,9 +884,9 @@ $(DO_X):
 	      done; \
 	      ;; \
 	    esac ; \
-	    export AR AS CC CXX LD NM RANLIB DLLTOOL WINDRES; \
+	    export AR ARFLAGS AS CC CXX LD NM RANLIB DLLTOOL WINDRES; \
 	    if (cd ./$$i; \
-	        $(MAKE) $(BASE_FLAGS_TO_PASS) "AR=$${AR}" "AS=$${AS}" \
+	        $(MAKE) $(BASE_FLAGS_TO_PASS) "AR=$${AR}" "ARFLAGS=$${ARFLAGS}" "AS=$${AS}" \
 			"CC=$${CC}" "CXX=$${CXX}" "LD=$${LD}" "NM=$${NM}" \
 			"RANLIB=$${RANLIB}" \
 			"DLLTOOL=$${DLLTOOL}" "WINDRES=$${WINDRES}" \
@@ -898,9 +903,9 @@ $(DO_X):
 	    for flag in $(EXTRA_TARGET_FLAGS); do \
 		eval `echo "$$flag" | sed -e "s|^\([^=]*\)=\(.*\)|\1='\2'|"`; \
 	    done; \
-	    export AR AS CC CXX LD NM RANLIB DLLTOOL WINDRES; \
+	    export AR ARFLAGS AS CC CXX LD NM RANLIB DLLTOOL WINDRES; \
 	    if (cd $(TARGET_SUBDIR)/$$i; \
-	        $(MAKE) $(BASE_FLAGS_TO_PASS) "AR=$${AR}" "AS=$${AS}" \
+	        $(MAKE) $(BASE_FLAGS_TO_PASS) "AR=$${AR}" "ARFLAGS=$${ARFLAGS}" "AS=$${AS}" \
 			"CC=$${CC}" "CXX=$${CXX}" "LD=$${LD}" "NM=$${NM}" \
 			"RANLIB=$${RANLIB}" \
 			"DLLTOOL=$${DLLTOOL}" "WINDRES=$${WINDRES}" \
@@ -911,10 +916,11 @@ $(DO_X):
 
 # Here are the targets which correspond to the do-X targets.
 
-.PHONY: info installcheck dvi install-info
+.PHONY: info guide installcheck dvi install-info install-guide
 .PHONY: clean distclean mostlyclean maintainer-clean realclean
 .PHONY: local-clean local-distclean local-maintainer-clean
 info: do-info
+guide: do-guide
 installcheck: do-installcheck
 dvi: do-dvi
 
@@ -927,6 +933,12 @@ install-info: do-install-info dir.info
 	  $(INSTALL_DATA) dir.info $(infodir)/dir.info ; \
 	else true ; fi
 
+install-guide: do-install-guide dir.guide
+	srcroot=`cd $(srcdir); pwd`; export srcroot; \
+	if [ -f dir.guide ] ; then \
+	  $(INSTALL_DATA) dir.guide $(guidedir)/dir.guide ; \
+	else true ; fi
+
 local-clean:
 	-rm -f *.a TEMP errs core *.o *~ \#* TAGS *.E
 
@@ -1350,7 +1362,7 @@ install-gcc:
 # This rule is used to install the modules which use FLAGS_TO_PASS.
 # To build a target install-X means to cd to X and make install.
 .PHONY: install-dosrel
-install-dosrel: installdirs info
+install-dosrel: installdirs info guide
 	@dir=`echo $@ | sed -e 's/install-//'`; \
 	if [ -f ./$${dir}/Makefile ] ; then \
 	  r=`pwd`; export r; \
@@ -1478,6 +1490,13 @@ dir.info: do-install-info
 	else true ; \
 	fi
 
+dir.guide: do-install-guide
+	if [ -f $(srcdir)/texinfo/gen-guide-dir ] ; then \
+	  $(srcdir)/texinfo/gen-guide-dir $(guidedir) $(srcdir)/texinfo/dir.guide-template > dir.guide.new ; \
+	  mv -f dir.guide.new dir.guide ; \
+	else true ; \
+	fi
+
 dist:
 	@echo "Building a full distribution of this tree isn't done"
 	@echo "via 'make dist'.  Check out the etc/ subdirectory" 
@@ -1513,7 +1532,7 @@ DEVO_SUPPORT= README Makefile.in configu
 # ChangeLog omitted because it may refer to files which are not in this
 # distribution (perhaps it would be better to include it anyway).
 ETC_SUPPORT= Makefile.in configure configure.in standards.texi \
-	make-stds.texi standards.info*
+	make-stds.texi standards.info* standards.guide*
 
 # When you use `make setup-dirs' or `make taz' you should always redefine
 # this macro.
