--- binutils-2.10.1/libiberty/argv.c	1999-07-14 19:32:01.000000000 +0200
+++ binutils-2.10.1/libiberty/argv.c	2012-08-08 12:35:51.000000000 +0200
@@ -40,7 +40,7 @@
 
 #else	/* !__STDC__ */
 
-#if !defined _WIN32 || defined __GNUC__
+#if !defined _WIN32 || !defined __APPLE__ || defined __GNUC__
 extern char *memcpy ();		/* Copy memory region */
 extern int strlen ();		/* Count length of string */
 extern char *malloc ();		/* Standard memory allocater */
