--- binutils-2.9.1/gas/write.h	Fri May  1 08:45:18 1998
+++ binutils-2.9.1/gas/write.h	Sun Aug 23 00:00:00 1998
@@ -177,17 +177,18 @@ extern void number_to_chars_bigendian   
 #ifdef BFD_ASSEMBLER
 extern fixS *fix_new
   PARAMS ((fragS * frag, int where, int size, symbolS * add_symbol,
-	   offsetT offset, int pcrel, bfd_reloc_code_real_type r_type));
+	   offsetT offset, int pcrel, bfd_reloc_code_real_type r_type,
+	   int baserel));
 extern fixS *fix_new_exp
   PARAMS ((fragS * frag, int where, int size, expressionS *exp, int pcrel,
-	   bfd_reloc_code_real_type r_type));
+	   bfd_reloc_code_real_type r_type, int baserel));
 #else
 extern fixS *fix_new
   PARAMS ((fragS * frag, int where, int size, symbolS * add_symbol,
-	   offsetT offset, int pcrel, int r_type));
+	   offsetT offset, int pcrel, int r_type, int baserel));
 extern fixS *fix_new_exp
   PARAMS ((fragS * frag, int where, int size, expressionS *exp, int pcrel,
-	   int r_type));
+	   int r_type, int baserel));
 #endif
 
 extern void write_print_statistics PARAMS ((FILE *));
