--- binutils-2.9.1/bfd/bfd-in.h	Fri May  1 08:48:01 1998
+++ binutils-2.9.1/bfd/bfd-in.h	Sun Aug 23 00:00:00 1998
@@ -694,4 +694,9 @@ extern boolean bfd_coff_get_syment
 extern boolean bfd_coff_get_auxent
   PARAMS ((bfd *, struct symbol_cache_entry *, int, union internal_auxent *));
 
+/* Externally visible PEI routines */
+
+extern void pei_get_comdat_info 
+  PARAMS ((struct sec *));
+
 /* And more from the source.  */
