--- binutils-2.9.1/bfd/hash.c	Fri May  1 08:48:10 1998
+++ binutils-2.9.1/bfd/hash.c	Sun Aug 23 00:00:00 1998
@@ -73,7 +73,7 @@ SUBSECTION
 	The function <<bfd_hash_table_init>> take as an argument a
 	function to use to create new entries.  For a basic hash
 	table, use the function <<bfd_hash_newfunc>>.  @xref{Deriving
-	a New Hash Table Type} for why you would want to use a
+	a New Hash Table Type}, for why you would want to use a
 	different value for this argument.
 
 @findex bfd_hash_allocate
