--- binutils-2.9.1/bfd/xcofflink.c	Fri May  1 08:48:17 1998
+++ binutils-2.9.1/bfd/xcofflink.c	Sun Aug 23 00:00:00 1998
@@ -721,7 +721,7 @@ _bfd_xcoff_canonicalize_dynamic_symtab (
 	    if (ldsym._l._l_name[i] == '\0')
 	      break;
 	  if (i < SYMNMLEN)
-	    symbuf->symbol.name = (char *) elsym->_l._l_name;
+	    symbuf->symbol.name = (const char *) elsym->_l._l_name;
 	  else
 	    {
 	      char *c;
@@ -3824,7 +3824,7 @@ xcoff_build_ldsyms (h, p)
 
       bfd_put_16 (ldinfo->output_bfd, len + 1,
 		  ldinfo->strings + ldinfo->string_size);
-      strcpy (ldinfo->strings + ldinfo->string_size + 2, h->root.root.string);
+      strcpy ((char *) ldinfo->strings + ldinfo->string_size + 2, h->root.root.string);
       h->ldsym->_l._l_l._l_zeroes = 0;
       h->ldsym->_l._l_l._l_offset = ldinfo->string_size + 2;
       ldinfo->string_size += len + 3;
@@ -3953,7 +3953,7 @@ _bfd_xcoff_bfd_final_link (abfd, info)
       if (file_align != 0)
 	{
 	  boolean saw_contents;
-	  int indx;
+	  int indx = 0;
 	  asection **op;
 	  file_ptr sofar;
 
