--- binutils-2.9.1/gprof/configure.in	Fri May  1 08:49:42 1998
+++ binutils-2.9.1/gprof/configure.in	Sun Aug 23 00:00:00 1998
@@ -28,6 +28,17 @@ case "${target}" in
   ;;
 esac
 
+### begin-GG-local
+# Check for additional host specific CFLAGS.
+AC_MSG_CHECKING(for host dependent CFLAGS)
+other_host_cflags=""
+case "${host}" in
+	m68*-*-amigaos*)	other_host_cflags="-mstackextend" ;;
+esac
+test -n "$other_host_cflags" && CFLAGS="$CFLAGS $other_host_cflags"
+AC_MSG_RESULT($other_host_cflags)
+### end-GG-local
+
 AM_MAINTAINER_MODE
 AM_CYGWIN32
 AM_EXEEXT
