--- binutils-2.9.1/gprof/configure	Fri May  1 08:49:42 1998
+++ binutils-2.9.1/gprof/configure	Thu Jan 21 19:39:59 1999
@@ -1521,8 +1901,20 @@ EOF
   ;;
 esac
 
+### begin-GG-local
+# Check for additional host specific CFLAGS.
+echo $ac_n "checking for host dependent CFLAGS""... $ac_c" 1>&6
+echo "configure:1908: checking for host dependent CFLAGS" >&5
+other_host_cflags=""
+case "${host}" in
+	m68*-*-amigaos*)	other_host_cflags="-mstackextend" ;;
+esac
+test -n "$other_host_cflags" && CFLAGS="$CFLAGS $other_host_cflags"
+echo "$ac_t""$other_host_cflags" 1>&6
+### end-GG-local
+
 echo $ac_n "checking whether to enable maintainer-specific portions of Makefiles""... $ac_c" 1>&6
 echo "configure:1526: checking whether to enable maintainer-specific portions of Makefiles" >&5
     # Check whether --enable-maintainer-mode or --disable-maintainer-mode was given.
 if test "${enable_maintainer_mode+set}" = set; then
   enableval="$enable_maintainer_mode"
