--- sfdc-1.4/main.pl	2005-09-24 16:49:49.000000000 +0200
+++ sfdc-1.4/main.pl	2013-08-30 15:13:28.000000000 +0200
@@ -418,7 +418,7 @@
     my $comment   = '';
 
     my $result = {
-	copyright  => 'Copyright © 2001 Amiga, Inc.',
+	copyright  => 'Copyright (c) 2001 Amiga, Inc.',
 	id         => '',
 	libname    => '',
 	base       => '',
@@ -849,8 +849,8 @@
 
 
     # Make sure we have the same number of arguments as registers, or,
-    # if this is a varargs function, possible one extra, á la "MethodID, ...".
-    # Tagcalls always have one extra, á la "Tag, ...".
+    # if this is a varargs function, possible one extra, a la "MethodID, ...".
+    # Tagcalls always have one extra, a la "Tag, ...".
 
     if (($prototype->{type} eq 'varargs' &&
 	 $prototype->{subtype} eq 'tagcall' &&
