--- sfdc-1.4/SFDCompile.pod	2005-09-24 16:49:49.000000000 +0200
+++ sfdc-1.4/SFDCompile.pod	2014-05-25 22:50:56.894529535 +0200
@@ -326,7 +326,7 @@
 =head1 NOTES
 
 Had I seen F<cvinclude.pl> before I started writing this program, I
-might still have been using fd/clib files. Or maybe not.
+might still have been using fd/clib files. Or maybe not.
 
 =head1 AUTHOR
 
@@ -334,98 +334,114 @@
 
 =head1 HISTORY
 
-=item B<1.0 (2003-07-27)>
+=over 4
+
+=item *
+
+B<1.0 (2003-07-27)>
 
 Initial release.
 
 
-=item B<1.1 (2003-12-22)>
+=item *
+
+B<1.1 (2003-12-22)>
 
 Added workaround for workbench.library (base name is "wb").
 Added AmigaOS 4 support.
 Added the B<--addvectors> switch.
 
 
-=item B<1.2 (2004-06-16)>
+=item *
+
+B<1.2 (2004-06-16)>
 
 Generates files for mathieeedoub*, though probably broken. Well they
 are the same as fd2inline generates at least.
 
 
-=item B<1.2a (2004-06-20)>
+=item *
+
+B<1.2a (2004-06-20)>
 
 Replace AROS_LP with AROS_LD. Because it is guaranteed to define the
 function prototype. [verhaegs]
 
-=item B<1.3 (2004-11-12)>
+=item *
+
+B<1.3 (2004-11-12)>
 
 Correctly handle the argument C<type **arg>, where there is no
 whitespace between the argument type and the argument name.
 
-=item
+=item *
 
 Correctly handle the prototype C<void function (void)>, were there is
 a whitespace between the function name and the parenthesis and "void"
 is used to indicate no arguments.
 
-=item
+=item *
 
 Replace C<-> with C<_> in base/library name.
 
-=item
+=item *
 
 New special keyword for register specification (in addition to B<sysv>
 and B<base>): B<autoreg>, which automatically allocates m68k registers
 for the arguments.
 
-=item
+=item *
 
 B<sysv> now works correctly with varargs functions. B<sysv> combined
 with B<gatestubs> work for m68k and i386 only for now. For AROS
 targets in B<macros>, the correct AROS macro is used to fetch the
 function to be called.
 
-=item
+=item *
 
 New B<--mode>: B<functable>. For proper code generation in AROS, make
 sure gateprotos are included before you include the functable.
 
-=item
+=item *
 
 B<proto> files now include F<Include/defines/> files when used in AROS.
 
-=item B<1.4 (2005-09-23)>
+=item *
+
+B<1.4 (2005-09-23)>
 
 Added AmigaOS 4-style interfaces (C++ only for non-OS4 targets, and
 without implementation).
 
-=item
+=item *
 
 The library base in the B<proto> file is now defined for AmigaOS 4
 too.
 
-=item
+=item *
 
 AmigaOS 4 B<clib> files now add B<__attribute__((linearvarargs))> to
 varargs functions.
 
-=item
+=item *
 
 Added AmigaOS 4 support for B<stubs> files.
 
-=item
+=item *
 
 Added support for the 'iptr' gcc attribute.
 
-=item
+=item *
 
 Added B<__SFDC_FUNCTABLE_M68K__> check in the B<functable> mode.
 
-=item
+=item *
 
 Added m68k stub functions to AmigaOS 4's B<gatestubs> and
 B<gateproto>.
 
-=item
+=item *
 
 Added the B<autoopen> mode.
+
+=back
