--- binutils-2.9.1/include/libiberty.h	1998-05-01 17:48:25.000000000 +0200
+++ binutils-2.9.1/include/libiberty.h	2012-08-04 12:43:35.000000000 +0200
@@ -16,6 +16,9 @@
 
 #include "ansidecl.h"
 
+#include <string.h>
+#include <stdlib.h>
+
 /* Build an argument vector from a string.  Allocates memory using
    malloc.  Use freeargv to free the vector.  */

@@ -111,12 +111,7 @@
 
 /* Exit, calling all the functions registered with xatexit.  */
 
-#ifndef __GNUC__
-extern void xexit PARAMS ((int status));
-#else
-typedef void libiberty_voidfn PARAMS ((int status));
-__volatile__ libiberty_voidfn xexit;
-#endif
+extern void xexit PARAMS ((int status));
 
