--- binutils-2.9.1/opcodes/ChangeLog	Fri May  1 08:48:18 1998
+++ binutils-2.9.1/opcodes/ChangeLog	Sun Aug 23 00:00:00 1998
@@ -333,6 +333,11 @@ Thu Oct 23 21:13:37 1997  Fred Fish  <fn
  
 	* d10v-opc.c (d10v_opcodes): Correct entry for RTE.
  
+Mon Oct 20 11:16:19 1997  Fred Fish  <fnf@ninemoons.com>
+
+	* configure.in (AC_ARFLAGS): Add.
+	* configure: Regenerate.
+
 Mon Dec  8 11:21:07 1997  Nick Clifton  <nickc@cygnus.com>
 
 	* disassemble.c: Remove disasm_symaddr() function.
