--- binutils-2.9.1/opcodes/configure.in	Fri May  1 08:48:19 1998
+++ binutils-2.9.1/opcodes/configure.in	Sun Aug 23 00:00:00 1998
@@ -18,6 +18,7 @@ AM_INIT_AUTOMAKE(opcodes, ${BFD_VERSION}
 dnl These must be called before AM_PROG_LIBTOOL, because it may want
 dnl to call AC_CHECK_PROG.
 AC_CHECK_TOOL(AR, ar)
+AC_ARFLAGS
 AC_CHECK_TOOL(RANLIB, ranlib, :)
 
 dnl Default to a non shared library.  This may be overridden by the
