--- binutils-2.9.1/config/mpw/g-mpw-make.sed	Fri May  1 08:48:45 1998
+++ binutils-2.9.1/config/mpw/g-mpw-make.sed	Sun Aug 23 00:00:00 1998
@@ -52,7 +52,7 @@
 /CC/s/^CC *=/#CC =/
 /CFLAGS/s/^CFLAGS *=/#CFLAGS =/
 /AR/s/^AR *=/#AR =/
-/AR_FLAGS/s/^AR_FLAGS *=/#AR_FLAGS =/
+/ARFLAGS/s/^ARFLAGS *=/#ARFLAGS =/
 /RANLIB/s/^RANLIB *=/#RANLIB =/
 /CC_LD/s/^CC_LD *=/#CC_LD =/
 /LDFLAGS/s/^LDFLAGS *=/#LDFLAGS =/
