--- binutils-2.14/bfd/bfd-in2.h	2003-05-03 18:06:57.000000000 +0200
+++ binutils-2.14-patched/bfd/bfd-in2.h	2013-04-24 18:38:25.000000000 +0200
@@ -3811,6 +3821,7 @@
       struct bfd_pef_data_struct *pef_data;
       struct bfd_pef_xlib_data_struct *pef_xlib_data;
       struct bfd_sym_data_struct *sym_data;
+      struct amiga_data_struct *amiga_data;
       PTR any;
     }
   tdata;
@@ -4069,6 +4080,7 @@
 enum bfd_flavour
 {
   bfd_target_unknown_flavour,
+  bfd_target_amiga_flavour,
   bfd_target_aout_flavour,
   bfd_target_coff_flavour,
   bfd_target_ecoff_flavour,
