--- binutils-2.14/bfd/targets.c	2003-04-01 17:50:27.000000000 +0200
+++ binutils-2.14-patched/bfd/targets.c	2013-04-24 18:38:23.000000000 +0200
@@ -138,6 +138,7 @@
 .enum bfd_flavour
 .{
 .  bfd_target_unknown_flavour,
+.  bfd_target_amiga_flavour,
 .  bfd_target_aout_flavour,
 .  bfd_target_coff_flavour,
 .  bfd_target_ecoff_flavour,
@@ -488,6 +489,8 @@
 extern const bfd_target a29kcoff_big_vec;
 extern const bfd_target a_out_adobe_vec;
 extern const bfd_target aix5coff64_vec;
+extern const bfd_target amiga_vec;
+extern const bfd_target aout_amiga_vec;
 extern const bfd_target aout0_big_vec;
 extern const bfd_target aout_arm_big_vec;
 extern const bfd_target aout_arm_little_vec;
@@ -763,6 +769,8 @@
 #ifdef BFD64
 	&aix5coff64_vec,
 #endif
+	&amiga_vec,
+	&aout_amiga_vec,
 	&aout0_big_vec,
 #if 0
 	/* We have no way of distinguishing these from other a.out variants.  */
