--- binutils-2.9.1/libiberty/ChangeLog	Fri May  1 08:47:47 1998
+++ binutils-2.9.1/libiberty/ChangeLog	Sun Aug 23 00:00:00 1998
@@ -129,6 +129,12 @@ Wed Sep 24 00:31:59 1997  Felix Lee  <fl
 	* config/mh-windows (EXTRA_OFILES): add asprintf.o and
  	strncasecmp.o.
 
+Mon Sep 22 14:26:22 1997  Fred Fish  <fnf@cygnus.com>
+
+	* Makefile.in (lneeded-list): Since xmalloc.o and xexit.o
+	are in REQUIRED_OFILES, there is no need to include them
+	again, so remove conditional include based on alloca.o use.
+
 Thu Aug 28 14:27:15 1997  Andrew Cagney  <cagney@b1.cygnus.com>
 
 	* vasprintf.c (vasprintf): Allow for _BSD_VA_LIST_.
