--- binutils-2.9.1/libiberty/config.table	1998-05-01 08:47:49.000000000 -0700
+++ binutils-2.9.1/libiberty/config.table	2011-05-08 20:28:36.636307945 -0700
@@ -23,6 +23,7 @@
   *-*-sysv*)		frag=mh-sysv ;;
   *-*-go32)		frag=mh-go32 ;;
   i[345]86-*-windows*)	frag=mh-windows ;;
+  *-*-amigaos*)		frag=mh-amigaos ;;
 
   *-*-vxworks5* | *-*-vxworks)
     # VxWorks 5 needs special action, because the usual
