--- binutils-2.9.1/libiberty/cplus-dem.c	1998-05-01 17:47:50.000000000 +0200
+++ binutils-2.9.1/libiberty/cplus-dem.c	2012-08-04 12:43:35.000000000 +0200
@@ -31,6 +31,7 @@
 #include <ctype.h>
 #include <string.h>
 #include <stdio.h>
+#include <stdlib.h>
 
 #include <demangle.h>
 #undef CURRENT_DEMANGLING_STYLE
@@ -3266,9 +3267,6 @@
   exit (1);
 }
 
-char * malloc ();
-char * realloc ();
-
 char *
 xmalloc (size)
      unsigned size;
