--- binutils-2.9.1/libiberty/Makefile.in	Fri May  1 08:47:47 1998
+++ binutils-2.9.1/libiberty/Makefile.in	Sun Aug 23 00:00:00 1998
@@ -28,7 +28,7 @@
 
 srcdir = .
 
-prefix = /usr/local
+prefix = @prefix@
 
 exec_prefix = $(prefix)
 bindir = $(exec_prefix)/bin
@@ -63,15 +63,15 @@ INSTALL = $(SHELL) $(srcdir)/../install-
 INSTALL_PROGRAM = $(INSTALL)
 INSTALL_DATA = $(INSTALL)
 
-AR = ar
-AR_FLAGS = rc
+AR = @AR@
+ARFLAGS = @ARFLAGS@
 
 ERRORS_CC = $(CC)
 CC = cc
 CFLAGS = -g
 LIBCFLAGS = $(CFLAGS)
 MAKEINFO = makeinfo
-RANLIB = ranlib
+RANLIB = @RANLIB@
 
 PICFLAG =
 
@@ -100,7 +100,7 @@ EXTRA_OFILES = 
 # Flags to pass to a recursive make.
 FLAGS_TO_PASS = \
 	"AR=$(AR)" \
-	"AR_FLAGS=$(AR_FLAGS)" \
+	"ARFLAGS=$(ARFLAGS)" \
 	"CC=$(CC)" \
 	"CFLAGS=$(CFLAGS)" \
 	"LIBCFLAGS=$(LIBCFLAGS)" \
@@ -130,7 +130,7 @@ check installcheck:
 
 INCDIR=$(srcdir)/$(MULTISRCTOP)../include
 
-COMPILE.c = $(CC) -c $(LIBCFLAGS) -I. -I$(INCDIR) $(HDEFINES)
+COMPILE.c = $(CC) -c $(LIBCFLAGS) -I. -I$(INCDIR) -I$(srcdir) $(HDEFINES)
 .c.o:
 	test -z "$(PICFLAG)" || \
 	  $(COMPILE.c) $(PICFLAG) $< -o pic/$@
@@ -204,9 +204,6 @@ lneeded-list: $(EXTRA_OFILES) needed.awk
 	    *) f="$$f $$i" ;; \
 	  esac ; \
 	done ; \
-	case $$f in \
-	    *alloca.o*) f="$$f xmalloc.o xexit.o" ;; \
-	esac ; \
 	echo $$f >>lneeded-list
 
 # Generate an awk script that looks for functions in functions.def
@@ -243,7 +240,7 @@ needed2.awk: $(srcdir)/functions.def Mak
 	 >>needed2.awk
 
 dummy.o: $(srcdir)/dummy.c $(srcdir)/functions.def
-	$(CC) -c $(CFLAGS) -I. -I$(INCDIR) $(HDEFINES) $(srcdir)/dummy.c 2>/dev/null
+	$(CC) -c $(CFLAGS) -I. -I$(INCDIR) -I$(srcdir) $(HDEFINES) $(srcdir)/dummy.c 2>/dev/null
 
 errors: dummy.o $(EXTRA_OFILES)
 	-($(ERRORS_CC) -o dummy $(CFLAGS) $(LDFLAGS) $(ERRORS_LDFLAGS) dummy.o $(EXTRA_OFILES) $(LOADLIBES)) >errors 2>&1 || true
@@ -263,7 +260,7 @@ $(RULE1): $(REQUIRED_OFILES) $(DO_ALSO) 
 RULE2 = not-used
 $(RULE2): $(REQUIRED_OFILES) $(HOST_OFILES)
 	rm -rf $(TARGETLIB)
-	$(AR) $(AR_FLAGS) $(TARGETLIB) \
+	$(AR) $(ARFLAGS) $(TARGETLIB) \
 	  $(REQUIRED_OFILES) $(HOST_OFILES)
 	$(RANLIB) $(TARGETLIB)
 
