--- binutils-2.9.1/bfd/archive.c	Fri May  1 08:48:01 1998
+++ binutils-2.9.1/bfd/archive.c	Sun Aug 23 00:00:00 1998
@@ -870,6 +870,22 @@ do_slurp_coff_armap (abfd)
     {
       rawptr = raw_armap + i;
       carsyms->file_offset = swap ((PTR) rawptr);
+      /* This is a hack to work around a bug in x86 BeOS linkers that did not
+	 follow the PE spec and saved the file offsets in little endian mode. */
+      {
+	file_ptr le_val;
+	le_val = bfd_getl32 ((PTR) rawptr);
+	if (carsyms->file_offset < 0)
+	  {
+	    /* Must have been little endian */
+	    carsyms->file_offset = le_val;
+	  }
+	else if ((le_val > 0) && (le_val < carsyms->file_offset))
+	  {
+	    /* Both values positive and le_val smaller, so use le_val. */
+	    carsyms->file_offset = le_val;
+	  }
+      }
       carsyms->name = stringbase;
       stringbase += strlen (stringbase) + 1;
       carsyms++;
