--- binutils-2.9.1/bfd/bfd.c	1998-05-01 17:48:01.000000000 +0200
+++ binutils-2.9.1/bfd/bfd.c	2012-08-04 12:43:35.000000000 +0200
@@ -177,6 +177,7 @@
 .      struct cisco_core_struct *cisco_core_data;
 .      struct versados_data_struct *versados_data;
 .      struct netbsd_core_struct *netbsd_core_data;
+.      struct amiga_data_struct *amiga_data;
 .      PTR any;
 .      } tdata;
 .  
@@ -1153,3 +1154,4 @@
 
   return true;
 }
+
