--- binutils-2.9.1/bfd/section.c	Fri May  1 08:48:15 1998
+++ binutils-2.9.1/bfd/section.c	Sun Aug 23 00:00:00 1998
@@ -150,6 +150,13 @@ SUBSECTION
 
 CODE_FRAGMENT
 .
+.typedef struct _comdat_info
+.{
+.  int asoc_sec;
+.  char name_buf[9];
+.  const char * comdat_sym;
+.} comdat_info;
+.
 .typedef struct sec
 .{
 .        {* The name of the section; the name isn't a copy, the pointer is
@@ -431,6 +438,8 @@ CODE_FRAGMENT
 .   struct symbol_cache_entry *symbol;
 .   struct symbol_cache_entry **symbol_ptr_ptr;
 .
+.   comdat_info *pe_comdat_info;
+.
 .   struct bfd_link_order *link_order_head;
 .   struct bfd_link_order *link_order_tail;
 .} asection ;
@@ -487,7 +496,7 @@ static const asymbol global_syms[] =
   const asection SEC = \
     { NAME, 0, 0, FLAGS, 0, 0, 0, 0, 0, 0, 0, 0, (asection *) &SEC, \
       0, 0, 0, 0, 0, 0, 0, 0,  0, 0, 0, 0, 0, 0, 0, 0, \
-      (asymbol *) &global_syms[IDX], (asymbol **) &SYM, 0, 0 }
+      (asymbol *) &global_syms[IDX], (asymbol **) &SYM, 0, 0, 0 }
 
 STD_SECTION (bfd_com_section, SEC_IS_COMMON, bfd_com_symbol,
 	     BFD_COM_SECTION_NAME, 0);
