--- binutils-2.9.1/bfd/syms.c	Fri May  1 08:48:15 1998
+++ binutils-2.9.1/bfd/syms.c	Sun Aug 23 00:00:00 1998
@@ -1062,7 +1062,7 @@ _bfd_stab_section_find_nearest_line (abf
     {
       stab = info->cached_stab;
       indexentry = info->cached_indexentry;
-      file_name = info->cached_file_name;
+      file_name = (char *) info->cached_file_name;
     }
   else
 #endif
@@ -1097,10 +1097,10 @@ _bfd_stab_section_find_nearest_line (abf
 	return true;
 
       stab = indexentry->stab + STABSIZE;
-      file_name = indexentry->file_name;
+      file_name = (char *) indexentry->file_name;
     }
 
-  directory_name = indexentry->directory_name;
+  directory_name = (char *) indexentry->directory_name;
   str = indexentry->str;
 
   for (; stab < (indexentry+1)->stab; stab += STABSIZE)
@@ -1188,11 +1188,11 @@ _bfd_stab_section_find_nearest_line (abf
          to clobber the colon.  It's OK to change the name, since the
          string is in our own local storage anyhow.  */
 
-      s = strchr (indexentry->function_name, ':');
+      s = strchr ((const char *) indexentry->function_name, ':');
       if (s != NULL)
 	*s = '\0';
 
-      *pfnname = indexentry->function_name;
+      *pfnname = (const char *) indexentry->function_name;
     }
 
   return true;
