--- gcc-2.95.3/gcc/c-tree.h	Thu Feb 18 21:38:43 1999
+++ gcc/gcc/c-tree.h	Thu Sep 27 19:35:53 2001
@@ -331,7 +331,9 @@
 extern void print_lang_type                     PROTO((FILE *, tree, int));
 extern void push_c_function_context             PROTO((void));
 extern void push_label_level                    PROTO((void));
-extern void push_parm_decl                      PROTO((tree));
+/* begin-GG-local: explicit register specification for parameters */
+extern void push_parm_decl                      PROTO((tree, tree));
+/* end-GG-local */
 extern tree pushdecl                            PROTO((tree));
 extern tree pushdecl_top_level                  PROTO((tree));
 extern void pushlevel                           PROTO((int));
