--- gcc-2.95.3/gcc/c-tree.h	1999-02-18 21:38:43.000000000 +0100
+++ gcc-2.95.3/gcc/c-tree.h	2012-08-04 11:53:28.000000000 +0200
@@ -331,7 +331,8 @@
 extern void print_lang_type                     PROTO((FILE *, tree, int));
 extern void push_c_function_context             PROTO((void));
 extern void push_label_level                    PROTO((void));
-extern void push_parm_decl                      PROTO((tree));
+/* Amiga: explicit register specification for parameters */
+extern void push_parm_decl                      PROTO((tree, tree));
 extern tree pushdecl                            PROTO((tree));
 extern tree pushdecl_top_level                  PROTO((tree));
 extern void pushlevel                           PROTO((int));
