--- gcc-2.95.3/gcc/configure.in	2001-01-25 06:03:02.000000000 -0800
+++ gcc-2.95.3/gcc/configure.in	2011-05-08 05:17:23.215286533 -0700
@@ -134,10 +134,11 @@
 elf="$with_elf",
 elf=no)
 
+### begin-GG-local: local prefix
 # Specify the local prefix
 local_prefix=
 AC_ARG_WITH(local-prefix,
-[  --with-local-prefix=DIR specifies directory to put local include.],
+[  --with-local-prefix=DIR specifies directory to put local include directory (not files).],
 [case "${withval}" in
 yes)	AC_MSG_ERROR(bad value ${withval} given for local include directory prefix) ;;
 no)	;;
@@ -146,8 +147,9 @@
 
 # Default local prefix if it is empty
 if test x$local_prefix = x; then
-	local_prefix=/usr/local
+	local_prefix='${prefix}'/local
 fi
+### end-GG-local
 
 # Don't set gcc_gxx_include_dir to gxx_include_dir since that's only
 # passed in by the toplevel make and thus we'd get different behavior
@@ -346,7 +348,7 @@
 AC_HEADER_TIME
 GCC_HEADER_STRING
 AC_HEADER_SYS_WAIT
-AC_CHECK_HEADERS(limits.h stddef.h string.h strings.h stdlib.h time.h fcntl.h unistd.h stab.h sys/file.h sys/time.h sys/resource.h sys/param.h sys/times.h sys/stat.h direct.h)
+AC_CHECK_HEADERS(limits.h stddef.h string.h strings.h stdlib.h syms.h time.h fcntl.h unistd.h stab.h sys/file.h sys/time.h sys/resource.h sys/param.h sys/times.h sys/stat.h direct.h varargs.h sys/varargs.h)
 
 # Check for thread headers.
 AC_CHECK_HEADER(thread.h, [have_thread_h=yes], [have_thread_h=])
@@ -416,6 +418,8 @@
 
 AC_DECL_SYS_SIGLIST
 
+AC_FUNC_ALLOCA
+
 # mkdir takes a single argument on some systems. 
 GCC_FUNC_MKDIR_TAKES_ONE_ARG
 
@@ -2013,6 +2017,14 @@
 		thread_file='vxworks'
 		float_format=m68k
 		;;
+	m68k-*-amigaos*)
+		xm_file=m68k/xm-amigaos.h
+		tm_file=m68k/amigaos.h
+		tmake_file=m68k/t-amigaos
+		xmake_file=m68k/x-amigaos
+		extra_objs=amigaos.o
+		gnu_ld=yes
+		;;
 	m68k-*-aout*)
 		tmake_file=m68k/t-m68kbare
 		tm_file="m68k/m68k-aout.h libgloss.h"
@@ -2118,7 +2130,6 @@
 		extra_headers=math-68881.h
 		float_format=m68k
 		;;
-
 	m88k-dg-dgux*)
 		case $machine in
 		  m88k-dg-dguxbcs*)
