--- gcc-2.95.3/gcc/configure	Fri Mar 16 15:13:48 2001
+++ gcc/gcc/configure	Wed Dec  5 23:55:03 2001
@@ -4347,6 +6190,14 @@
 		extra_headers=math-68881.h
 		float_format=m68k
 		;;
+	m68k-*-amigaos*)
+		xm_file=m68k/xm-amigaos.h
+		tm_file=m68k/amigaos.h
+		tmake_file=m68k/t-amigaos
+		xmake_file=m68k/x-amigaos
+		extra_objs=amigaos.o
+		gnu_ld=yes
+		;;
 	m68k-*-coff*)
 		tmake_file=m68k/t-m68kbare
 		tm_file="m68k/m68k-coff.h dbx.h libgloss.h"
