--- gcc-2.95.3/gcc/config.in	2001-03-16 15:09:04.000000000 +0100
+++ gcc-2.95.3/gcc/config.in	2012-08-04 11:53:28.000000000 +0200
@@ -194,6 +194,9 @@
 /* Define vfork as fork if vfork does not work.  */
 #undef vfork
 
+/* Define if vfork() works */
+#undef HAVE_VFORK
+
 /* Define if you have the __argz_count function.  */
 #undef HAVE___ARGZ_COUNT
 
@@ -338,6 +341,9 @@
 /* Define if you have the <strings.h> header file.  */
 #undef HAVE_STRINGS_H
 
+/* Define if you have the <syms.h> header file.  */
+#undef HAVE_SYMS_H
+
 /* Define if you have the <sys/file.h> header file.  */
 #undef HAVE_SYS_FILE_H
 
@@ -356,11 +362,17 @@
 /* Define if you have the <sys/times.h> header file.  */
 #undef HAVE_SYS_TIMES_H
 
+/* Define if you have the <sys/varargs.h> header file.  */
+#undef HAVE_SYS_VARARGS_H
+
 /* Define if you have the <time.h> header file.  */
 #undef HAVE_TIME_H
 
 /* Define if you have the <unistd.h> header file.  */
 #undef HAVE_UNISTD_H
 
+/* Define if you have the <varargs.h> header file.  */
+#undef HAVE_VARARGS_H
+
 /* Define if you have the i library (-li).  */
 #undef HAVE_LIBI
