--- gcc-2.95.3/gcc/configure.in	Thu Jan 25 15:03:02 2001
+++ gcc/gcc/configure.in	Wed Dec  5 23:54:53 2001
@@ -2019,6 +2038,14 @@
 		extra_headers=math-68881.h
 		float_format=m68k
 		;;
+	m68k-*-amigaos*)
+		xm_file=m68k/xm-amigaos.h
+		tm_file=m68k/amigaos.h
+		tmake_file=m68k/t-amigaos
+		xmake_file=m68k/x-amigaos
+		extra_objs=amigaos.o
+		gnu_ld=yes
+		;;
 	m68k-*-coff*)
 		tmake_file=m68k/t-m68kbare
 		tm_file="m68k/m68k-coff.h dbx.h libgloss.h"
@@ -2763,6 +2790,16 @@
 		tmake_file=rs6000/t-beos
 		xmake_file=rs6000/x-beos
 		;;
+	powerpc-*-morphos*)
+		cpu_type=rs6000
+		xm_file=rs6000/xm-morphos.h
+		tm_file=rs6000/morphos.h
+		tmake_file=rs6000/t-morphos
+		xmake_file=rs6000/x-morphos
+		extra_objs=morphos.o
+		broken_install=cp
+		install_headers_dir=install-headers-cp
+		;;
 	powerpc-*-sysv* | powerpc-*-elf*)
 		tm_file=rs6000/sysv4.h
 		xm_file="xm-siglist.h rs6000/xm-sysv4.h"
