--- gcc-2.95.3/gcc/configure	2001-03-16 15:09:04.000000000 +0100
+++ gcc-2.95.3/gcc/configure	2012-08-04 11:53:28.000000000 +0200
@@ -4341,6 +4352,13 @@
 		thread_file='vxworks'
 		float_format=m68k
 		;;
+	m68k-*-amigaos*)
+		tm_file=m68k/amigaos.h
+		tmake_file=m68k/t-amigaos
+		xmake_file=m68k/x-amigaos
+		extra_objs=amigaos.o
+		gnu_ld=yes
+		;;
 	m68k-*-aout*)
 		tmake_file=m68k/t-m68kbare
 		tm_file="m68k/m68k-aout.h libgloss.h"
