--- gcc-2.95.3/gcc/configure.in	2001-01-25 15:03:02.000000000 +0100
+++ gcc-2.95.3/gcc/configure.in	2012-08-04 11:53:28.000000000 +0200
@@ -134,10 +134,11 @@
 elf="$with_elf",
 elf=no)
 
+### begin-GG-local: local prefix
 # Specify the local prefix
 local_prefix=
 AC_ARG_WITH(local-prefix,
-[  --with-local-prefix=DIR specifies directory to put local include.],
+[  --with-local-prefix=DIR specifies directory to put local include directory (not files).],
 [case "${withval}" in
 yes)	AC_MSG_ERROR(bad value ${withval} given for local include directory prefix) ;;
 no)	;;
@@ -146,8 +147,9 @@
 
 # Default local prefix if it is empty
 if test x$local_prefix = x; then
-	local_prefix=/usr/local
+	local_prefix='${prefix}'/local
 fi
+### end-GG-local
 
 # Don't set gcc_gxx_include_dir to gxx_include_dir since that's only
 # passed in by the toplevel make and thus we'd get different behavior
@@ -2013,6 +2017,13 @@
 		thread_file='vxworks'
 		float_format=m68k
 		;;
+	m68k-*-amigaos*)
+		tm_file=m68k/amigaos.h
+		tmake_file=m68k/t-amigaos
+		xmake_file=m68k/x-amigaos
+		extra_objs=amigaos.o
+		gnu_ld=yes
+		;;
 	m68k-*-aout*)
 		tmake_file=m68k/t-m68kbare
 		tm_file="m68k/m68k-aout.h libgloss.h"
