--- gcc-2.95.3/gcc/genconfig.c	1999-04-16 12:52:24.000000000 -0700
+++ gcc-2.95.3/gcc/genconfig.c	2011-05-08 04:56:30.295286222 -0700
@@ -327,8 +327,8 @@
   printf ("/* Generated automatically by the program `genconfig'\n\
 from the machine description file `md'.  */\n\n");
 
-  /* Allow at least 10 operands for the sake of asm constructs.  */
-  max_recog_operands = 9;  /* We will add 1 later.  */
+  /* Allow at least 15 operands for the sake of asm constructs.  */
+  max_recog_operands = 14;  /* We will add 1 later.  */
   max_dup_operands = 1;
 
   /* Read the machine description.  */
