--- gcc-2.95.3/gcc/protoize.c	1999-04-16 21:52:36.000000000 +0200
+++ gcc-2.95.3/gcc/protoize.c	2012-08-04 11:53:28.000000000 +0200
@@ -255,6 +255,7 @@
     { CROSS_INCLUDE_DIR, 0, 0, 0 },
     /* This is another place that the target system's headers might be.  */
     { TOOL_INCLUDE_DIR, "BINUTILS", 0, 0 },
+    { LOCAL_INCLUDE_DIR, 0, 0, 1 },
 #else /* not CROSS_COMPILE */
     /* This should be /use/local/include and should come before
        the fixincludes-fixed header files.  */
