--- gcc-2.95.3/gcc/reload.c	Thu Jan 25 15:03:20 2001
+++ gcc/gcc/reload.c	Fri Sep 28 13:21:57 2001
@@ -2125,13 +2125,17 @@
       /* On a WORDS_BIG_ENDIAN machine, point to the last register of a
 	 multiple hard register group, so that for example (reg:DI 0) and
 	 (reg:SI 1) will be considered the same register.  */
-      if (WORDS_BIG_ENDIAN && GET_MODE_SIZE (GET_MODE (x)) > UNITS_PER_WORD
+
+      if (((!FLOAT_MODE_P (GET_MODE (x)) && WORDS_BIG_ENDIAN) ||
+	   ( FLOAT_MODE_P (GET_MODE (x)) && FLOAT_WORDS_BIG_ENDIAN))
+	  && GET_MODE_SIZE (GET_MODE (x)) > UNITS_PER_WORD
 	  && i < FIRST_PSEUDO_REGISTER)
 	i += (GET_MODE_SIZE (GET_MODE (x)) / UNITS_PER_WORD) - 1;
-      if (WORDS_BIG_ENDIAN && GET_MODE_SIZE (GET_MODE (y)) > UNITS_PER_WORD
+      if (((!FLOAT_MODE_P (GET_MODE (y)) && WORDS_BIG_ENDIAN) ||
+	   ( FLOAT_MODE_P (GET_MODE (y)) && FLOAT_WORDS_BIG_ENDIAN))
+	  && GET_MODE_SIZE (GET_MODE (y)) > UNITS_PER_WORD
 	  && j < FIRST_PSEUDO_REGISTER)
 	j += (GET_MODE_SIZE (GET_MODE (y)) / UNITS_PER_WORD) - 1;
-
       return i == j;
     }
   /* If two operands must match, because they are really a single
