--- gcc-2.95.3/gcc/sdbout.c	1999-04-18 15:09:29.000000000 +0200
+++ gcc-2.95.3/gcc/sdbout.c	2012-08-04 11:53:28.000000000 +0200
@@ -59,7 +59,7 @@
    supply usable syms.h include files.  Which syms.h file to use is a
    target parameter so don't use the native one if we're cross compiling.  */
 
-#if defined(USG) && !defined(MIPS) && !defined (hpux) && !defined(_WIN32) && !defined(__linux__) && !defined(__INTERIX) && !defined(CROSS_COMPILE)
+#if defined (HAVE_SYMS_H) && defined(USG) && !defined(MIPS) && !defined (hpux) && !defined(_WIN32) && !defined(__linux__) && !defined(__INTERIX) && !defined(CROSS_COMPILE)
 #include <syms.h>
 /* Use T_INT if we don't have T_VOID.  */
 #ifndef T_VOID
