--- gcc-2.95.3/gcc/toplev.c	2001-01-25 15:03:23.000000000 +0100
+++ gcc-2.95.3/gcc/toplev.c	2012-08-04 11:53:28.000000000 +0200
@@ -924,6 +924,10 @@
   {"pic", &flag_pic, 1,
    "Generate position independent code, if possible"},
   {"PIC", &flag_pic, 2, ""},
+  {"baserel", &flag_pic, 3,
+   "Generate base relative code"},
+  {"baserel32", &flag_pic, 4,
+   "Generate base relative code with no data limits"},
   {"exceptions", &flag_exceptions, 1,
    "Enable exception handling" },
   {"new-exceptions", &flag_new_exceptions, 1,
