--- binutils-2.9.1/ld/configure.tgt	Fri May  1 08:48:48 1998
+++ binutils-2.9.1/ld/configure.tgt	Sun Aug 23 00:00:00 1998
@@ -119,6 +120,7 @@ sh-*-*|sh-*-rtems*)	targ_emul=sh; targ_e
 m68k-sony-*)		targ_emul=news ;;
 m68k-hp-bsd*)		targ_emul=hp300bsd ;;
 m68*-motorola-sysv*)	targ_emul=delta68 ;;
+m68*-*-amigaos*)	targ_emul=amiga ; targ_extra_emuls=amiga_bss;;
 m68*-*-aout)		targ_emul=m68kaout ;;
 m68*-*-coff)		targ_emul=m68kcoff ;;
 m68*-*-elf)		targ_emul=m68kelf ;;
@@ -180,6 +182,8 @@ alpha*-*-netbsd*)	targ_emul=elf64alpha ;
 z8k-*-coff) 		targ_emul=z8002; targ_extra_emuls=z8001 ;;
 ns32k-pc532-mach* | ns32k-pc532-ux*)	targ_emul=pc532macha ;;
 ns32k-pc532-netbsd* | ns32k-pc532-lites*) targ_emul=ns32knbsd ;;
+powerpc-*-amigaos*)	targ_emul=elf32ppc ;;
+powerpc-*-amigaoshunk*)	targ_emul=ppcamiga ; targ_extra_emuls=ppcamiga_bss;;
 powerpc-*-elf* | powerpc-*-eabi* | powerpc-*-linux-gnu* | powerpc-*-sysv* \
   | powerpc-*-netbsd*)
 	targ_emul=elf32ppc ;;
