--- binutils-2.9.1/ld/configure.host	Fri May  1 08:48:48 1998
+++ binutils-2.9.1/ld/configure.host	Sun Aug 23 00:00:00 1998
@@ -35,6 +35,10 @@ alpha*-*-*)
   NATIVE_LIB_DIRS=/usr/ccs/lib
   ;;
 
+i[3456]86-*-beos*)
+  NATIVE_LIB_DIRS=/boot/develop/lib/x86
+  ;;
+
 i[3456]86-*-bsd* | i[3456]86-*-freebsd* | i[3456]86-*-netbsd*)
   # The new BSD `make' has a bug: it doesn't pass empty arguments in
   # shell commands.  So we need to make this value non-empty in order
@@ -112,6 +116,10 @@ mips*-sgi-irix4*)
 mips*-sgi-irix[56]*)
   HOSTING_CRT0=/usr/lib/crt1.o
   HOSTING_LIBS='`if [ -f ../gcc/libgcc.a ] ; then echo ../gcc/libgcc.a ; else gcc -print-libgcc-file-name; fi` -lc /usr/lib/crtn.o'
+  ;;
+
+m68*-*-amigaos*)
+  HDEFINES=-mstackextend
   ;;
 
 m68*-*-linux*aout*)
