--- gcc-2.95.3/libiberty/strsignal.c	1998-05-16 01:42:42.000000000 +0200
+++ gcc-2.95.3/libiberty/strsignal.c	2012-08-04 11:53:28.000000000 +0200
@@ -25,8 +25,10 @@
 
 #ifdef __STDC__
 #include <stddef.h>
+#ifndef __APPLE__
 extern void *malloc (size_t size);				/* 4.10.3.3 */
 extern void *memset (void *s, int c, size_t n);			/* 4.11.6.1 */
+#endif
 #else	/* !__STDC__ */
 extern char *malloc ();		/* Standard memory allocater */
 extern char *memset ();
@@ -584,8 +586,8 @@
 
 void
 psignal (signo, message)
-  unsigned signo;
-  char *message;
+  unsigned int signo;
+  const char *message;
 {
   if (signal_names == NULL)
     {
