--- binutils-2.9.1/binutils/dlltool.c	Fri May  1 08:49:32 1998
+++ binutils-2.9.1/binutils/dlltool.c	Sun Aug 23 00:00:00 1998
@@ -314,7 +314,7 @@ static const unsigned char arm_jtab[] =
 static const unsigned char ppc_jtab[] = 
 { 
   0x00, 0x00, 0x62, 0x81, /* lwz r11,0(r2)               */
-                          /*   Reloc TOCREL16 __imp_xxx  */
+                          /*   Reloc TOCREL16 _imp__xxx  */
   0x00, 0x00, 0x8B, 0x81, /* lwz r12,0(r11)              */
   0x04, 0x00, 0x41, 0x90, /* stw r2,4(r1)                */
   0xA6, 0x03, 0x89, 0x7D, /* mtctr r12                   */
@@ -1292,14 +1292,14 @@ both single and double underscores, for 
 
 	.text
 	.global	_GetFileVersionInfoSizeW@8
-	.global	__imp_GetFileVersionInfoSizeW@8
+	.global	_imp__GetFileVersionInfoSizeW@8
 _GetFileVersionInfoSizeW@8:
-	jmp *	__imp_GetFileVersionInfoSizeW@8
+	jmp *	_imp__GetFileVersionInfoSizeW@8
 	.section	.idata$7	# To force loading of head
 	.long	__version_a_head
 # Import Address Table
 	.section	.idata$5
-__imp_GetFileVersionInfoSizeW@8:
+_imp__GetFileVersionInfoSizeW@8:
 	.rva	ID2
 
 # Import Lookup Table
@@ -1316,8 +1316,8 @@ For the PowerPC, here's the variation on
 # Rather than a simple "jmp *", the code to get to the dll function 
 # looks like:
          .text
-         lwz	r11,[tocv]__imp_function_name(r2)
-#		   RELOC: 00000000 TOCREL16,TOCDEFN __imp_function_name
+         lwz	r11,[tocv]_imp__function_name(r2)
+#		   RELOC: 00000000 TOCREL16,TOCDEFN _imp__function_name
          lwz	r12,0(r11)
 	 stw	r2,4(r1)
 	 mtctr	r12
@@ -1353,7 +1353,7 @@ make_one_lib_file (exp, i)
       fprintf (f, "\t%s\t%s%s\n", ASM_GLOBAL, ASM_PREFIX, exp->name);
       fprintf (f, "\t%s\t__imp_%s\n", ASM_GLOBAL, exp->name);
       fprintf (f, "\t%s\t_imp__%s\n", ASM_GLOBAL, exp->name);
-      fprintf (f, "%s%s:\n\t%s\t__imp_%s\n", ASM_PREFIX,
+      fprintf (f, "%s%s:\n\t%s\t_imp__%s\n", ASM_PREFIX,
 	       exp->name, ASM_JUMP, exp->name);
 
       fprintf (f, "\t.section\t.idata$7\t%s To force loading of head\n", ASM_C);
